modules/apps/portal-cache/portal-cache-impl/src/main/java/com/liferay/portal/cache/internal/dao/orm/EntityCacheListener.java

```diff
@@ -0,0 +1,166 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.portal.cache.internal.dao.orm;
+
+import com.liferay.osgi.service.tracker.collections.map.ServiceTrackerMap;
+import com.liferay.portal.kernel.dao.orm.FinderCache;
+import com.liferay.portal.kernel.dao.orm.FinderPath;
+import com.liferay.portal.kernel.model.BaseModel;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * @author Tina Tian
+ */
+public class EntityCacheListener {
```

Comment:
This EntityCacheListener seems a bit pointless. Why not just making EntityCacheImpl having a reference to FinderCache?

Reply:
@shuyangzhou FinderCache depends on EntityCache, if use a direct reference, it would be a circle.

Reply:
Not by @Reference, but a real reference setting, same idea of this listener, just do it literally, This listener is not really a listener, just granting a indirect access to FinderCache. We should make this more explicit.

Reply:
A question is: do we want it to be coupled with EntityCache?
Currently the update of finder cache depending on entity cache, you can not clean up your finder cache directly. 
I am not sure if CT will need it later, but as I know, we want to make it something more independent, maybe like MessageBus, right?
My plan is to extract more common api to update finder cache, and will move the logic in the listener to the impl of API, there will be no listener.

Reply:
You are not supposed to cleanup finder cache directly. It is driven by entity changes. Not on callers' will. But, yeah, decoupling can always be good, just it is not really needed at this moment. We can deal with it a bit later.

Reply:
OK, I see..